### ISSUE TITLE
ICE failed for component 1 in stream 1...

### ISSUE THREAD

**Issue by mohammadhasanzadeh** [canyanio/janus-gateway-docker#16]
Hello,
I get the `ICE failed for component 1 in stream 1...` error on the local network,
My container IP is `172.18.0.2` and my host IP is `192.168.8.100`,
I also tested with STUN and TURN server, but error still occurs.
it seems is a docker container networking error, Any ideas for this ?

**Comment by mohammadhasanzadeh** [comment]
I fixed this issue by adding `nat_1_1_mapping = "192.168.8.101` in janus config file.
I'm not sure this is the correct way, but that works.

**Comment by michkish** [comment]
I think that because janus-gateway is in docker container, the program identifies 172.18.X.X is local ip(172.18.X.X is docker inner ip).

**Comment by michkish** [comment]
this bug I think should resign to janus dev team, janus program did not know it was in a docker container. Your suggestion also helped me, and I suggest try set network_mode to host also can fix the bug if you use docker-compose.

**Comment by mohammadhasanzadeh** [comment]
Yes, for my case, both of the solutions works, but I use a second workaround (`network_mode`).

**Comment by alexthedamager** [comment]
`nat_1_1_mapping` didn't work for me but setting `network_mode` to `host` did work - though I'm not sure why.

**Comment by alexberezin** [comment]
Hello! Other thing worked for me - I set STUN server. It turns out, that it's important for docker, because it uses NAT.
Set this parameters in janus.jcfg:
stun_server = "stun.l.google.com"
stun_port = 19302
